dcos-log/api/v2/handlers.go

```diff
@@ -0,0 +1,293 @@
+
+	nodeInfo, ok := middleware.FromContextNodeInfo(req.Context())
+	if !ok {
+		ERROR(w, "invalid context, unable to retrieve a nodeInfo object", http.StatusInternalServerError)
+		return
+	}
+
+	scheme := "http"
+	if cfg.FlagAuth {
+		scheme = "https"
+	}
+
+	vars := mux.Vars(req)
+	frameworkID := vars["frameworkID"]
+	executorID := vars["executorID"]
+	containerID := vars["containerID"]
+	taskPath := vars["taskPath"]
+	file := vars["file"]
+
+	token, ok := middleware.FromContextToken(req.Context())
+	if !ok {
+		ERROR(w, "unable to get authorization header from a request", http.StatusUnauthorized)
+		return
+	}
+
+	header := http.Header{}
+	header.Set("Authorization", token)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	mesosID, err := nodeInfo.MesosID(nodeutil.NewContextWithHeaders(ctx, header))
+	if err != nil {
+		ERROR(w, "unable to get mesosID: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	ip, err := nodeInfo.DetectIP()
+	if err != nil {
+		ERROR(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	masterURL := url.URL{
+		Scheme: scheme,
+		Host:   net.JoinHostPort(ip.String(), strconv.Itoa(dcos.PortMesosAgent)),
+		Path:   "/files/read",
+	}
+
+	opts := []reader.Option{reader.OptHeaders(header)}
+
+	var limit int
+	if limitStr := req.URL.Query().Get("limit"); limitStr != "" {
+		limit, err = strconv.Atoi(limitStr)
+		if err != nil {
+			ERROR(w, "unable to parse integer "+limitStr, http.StatusBadRequest)
+			return
+		}
+	}
+
+	if limit > 0 {
+		opts = append(opts, reader.OptLines(limit))
+	}
+
+	cursor := -1
+	if cursorStr := req.URL.Query().Get("cursor"); cursorStr != "" {
+		if cursorStr == "BEG" {
+			cursor = 0
+		} else if cursorStr == "END" {
+			opts = append(opts, reader.OptReadFromEnd())
+		} else {
+			cursor, err = strconv.Atoi(cursorStr)
+			if err != nil {
+				ERROR(w, "unable to parse integer "+cursorStr, http.StatusBadRequest)
+				return
+			}
+
+			if cursor >= 0 {
+				opts = append(opts, reader.OptOffset(cursor))
+			}
+		}
+	}
+
+	var skip int
+	if skipStr := req.URL.Query().Get("skip"); skipStr != "" {
+		skip, err = strconv.Atoi(skipStr)
+		if err != nil {
+			ERROR(w, "unable to parse integer "+skipStr, http.StatusBadRequest)
+			return
+		}
+
+		if skip != 0 {
+			opts = append(opts, reader.OptSkip(skip))
+		}
+
+		if skip < 0 {
+			opts = append(opts, reader.OptReadDirection(reader.BottomToTop))
+		}
+	}
+
+	lastEventID := req.Header.Get("Last-Event-ID")
+	if lastEventID != "" {
+		offset, err := strconv.Atoi(lastEventID)
+		if err != nil {
+			ERROR(w, fmt.Sprintf("invalid Last-Event-ID: %s", err.Error()), http.StatusInternalServerError)
+			return
+		}
+
+		opts = append(opts, reader.OptOffset(offset))
+	}
+
+	defaultFormatter := reader.LineFormat
+	stream := false
+	if req.Header.Get("Accept") == "text/event-stream" {
+		defaultFormatter = reader.SSEFormat
+		stream = true
+	}
+
+	opts = append(opts, reader.OptStream(stream))
```

Comment:
Do we need stream := false outside the if block above?  Could we do this instead?

```
stream := false // remove this
if req.Header.Get("Accept") == "text/event-stream" {
    defaultFormatter = reader.SSEFormat
    opts = append(opts, reader.OptStream(true))
}
```